
* stuff is in the wrong directory, move to sbin
* CFLAGS is abused for preprocessor, move to CPPFLAGS
* add missing $(LIBGTPNL_CFLAGS)
* ggsn includes libmnl.h but fails to PKG_CHECK for it;
  turns out it only uses the header and no defs, so kill that.
* _DEPENDENCIES on files not generated by $this Makefile are pointless
* must use .la files in _LDADD/_LIBADD whenever they exist
* remove AC_PROG_CXX, the C++ is never used

---
 ggsn/Makefile.am      |   16 +++++++++-------
 gtp/Makefile.am       |    5 +----
 lib/Makefile.am       |   16 +++++++++-------
 lib/gtp-kernel.c      |    1 +
 sgsnemu/Makefile.am   |   14 ++++++++------
 tests/gtp/Makefile.am |    2 +-
 tests/lib/Makefile.am |    4 ++--
 7 files changed, 31 insertions(+), 27 deletions(-)

Index: osmo-ggsn-1.15.0/ggsn/Makefile.am
===================================================================
--- osmo-ggsn-1.15.0.orig/ggsn/Makefile.am
+++ osmo-ggsn-1.15.0/ggsn/Makefile.am
@@ -1,11 +1,9 @@
-bin_PROGRAMS = osmo-ggsn
+sbin_PROGRAMS = osmo-ggsn
 
 AM_LDFLAGS = @EXEC_LDFLAGS@
 
-AM_CFLAGS = \
+AM_CPPFLAGS = \
 	    -D_GNU_SOURCE \
-	    -fno-builtin \
-	    -Wall \
 	    -DSBINDIR='"$(sbindir)"' \
 	    -I$(top_srcdir)/include \
 	    $(LIBOSMOCORE_CFLAGS) \
@@ -14,9 +12,13 @@ AM_CFLAGS = \
 	    $(LIBOSMONETIF_CFLAGS) \
 	    $(NULL)
 
+AM_CFLAGS =
+	    -fno-builtin \
+	    -Wall
+
 osmo_ggsn_LDADD = \
 	@EXEC_LDADD@ \
-	-lgtp -L../gtp ../lib/libmisc.a \
+	-lgtp -L../gtp ../lib/libmisc.la \
 	$(LIBOSMOCORE_LIBS) \
 	$(LIBOSMOCTRL_LIBS) \
 	$(LIBOSMOVTY_LIBS) \
@@ -24,9 +26,9 @@ osmo_ggsn_LDADD = \
 	$(NULL)
 
 if ENABLE_GTP_KERNEL
-AM_CFLAGS += -DGTP_KERNEL $(LIBGTPNL_CFLAGS)
+AM_CPPFLAGS += -DGTP_KERNEL $(LIBGTPNL_CFLAGS)
 osmo_ggsn_LDADD += $(LIBGTPNL_LIBS)
 endif
 
-osmo_ggsn_DEPENDENCIES = ../gtp/libgtp.la ../lib/libmisc.a
+osmo_ggsn_DEPENDENCIES = ../gtp/libgtp.la ../lib/libmisc.la
 osmo_ggsn_SOURCES = ggsn_main.c ggsn_vty.c ggsn.c ggsn.h sgsn.c sgsn.h pco.c pco.h
Index: osmo-ggsn-1.15.0/gtp/Makefile.am
===================================================================
--- osmo-ggsn-1.15.0.orig/gtp/Makefile.am
+++ osmo-ggsn-1.15.0/gtp/Makefile.am
@@ -6,13 +6,10 @@ LIBVERSION=12:1:1
 
 lib_LTLIBRARIES = libgtp.la
 
+AM_CPPFLAGS = -DSBINDIR='"$(sbindir)"' -I$(top_srcdir)/include $(LIBOSMOCORE_CFLAGS) ${LIBOSMOGSM_CFLAGS}
 AM_CFLAGS = \
 	    -fno-builtin \
 	    -Wall \
-	    -DSBINDIR='"$(sbindir)"' \
-	    -I$(top_srcdir)/include \
-	    $(LIBOSMOCORE_CFLAGS) \
-	    $(LIBOSMOGSM_CFLAGS) \
 	    $(NULL)
 
 libgtp_la_SOURCES = \
Index: osmo-ggsn-1.15.0/lib/Makefile.am
===================================================================
--- osmo-ggsn-1.15.0.orig/lib/Makefile.am
+++ osmo-ggsn-1.15.0/lib/Makefile.am
@@ -1,4 +1,4 @@
-noinst_LIBRARIES = libmisc.a
+noinst_LTLIBRARIES = libmisc.la
 
 noinst_HEADERS = \
 		 gnugetopt.h \
@@ -14,16 +14,18 @@ noinst_HEADERS = \
 		 util.h \
 		 $(NULL)
 
-AM_CFLAGS = \
-	    -fno-builtin \
-	    -Wall \
+AM_CPPFLAGS = \
 	    -DSBINDIR='"$(sbindir)"' \
 	    -I$(top_srcdir)/include \
 	    $(LIBOSMOCORE_CFLAGS) \
 	    $(LIBOSMONETIF_CFLAGS) \
 	    $(NULL)
 
-libmisc_a_SOURCES = \
+AM_CFLAGS = \
+	    -fno-builtin \
+	    -Wall
+
+libmisc_la_SOURCES = \
 		    debug.c \
 		    getopt.c \
 		    getopt1.c \
@@ -38,6 +40,6 @@ libmisc_a_SOURCES = \
 		    $(NULL)
 
 if ENABLE_GTP_KERNEL
-AM_CFLAGS += -DGTP_KERNEL $(LIBGTPNL_CFLAGS)
-libmisc_a_SOURCES += gtp-kernel.c
+AM_CPPFLAGS += -DGTP_KERNEL $(LIBGTPNL_CFLAGS)
+libmisc_la_SOURCES += gtp-kernel.c
 endif
Index: osmo-ggsn-1.15.0/lib/gtp-kernel.c
===================================================================
--- osmo-ggsn-1.15.0.orig/lib/gtp-kernel.c
+++ osmo-ggsn-1.15.0/lib/gtp-kernel.c
@@ -54,6 +54,7 @@ static void pdp_debug(const char *prefix
 		buf4, buf6, inet_ntoa(ia));
 }
 
+struct mnl_socket;
 static struct {
 	int			genl_id;
 	struct mnl_socket	*nl;
Index: osmo-ggsn-1.15.0/sgsnemu/Makefile.am
===================================================================
--- osmo-ggsn-1.15.0.orig/sgsnemu/Makefile.am
+++ osmo-ggsn-1.15.0/sgsnemu/Makefile.am
@@ -2,30 +2,32 @@ bin_PROGRAMS = sgsnemu
 
 AM_LDFLAGS = @EXEC_LDFLAGS@
 
-AM_CFLAGS = \
+AM_CPPFLAGS = \
 	    -D_GNU_SOURCE \
-	    -fno-builtin \
-	    -Wall \
 	    -DSBINDIR='"$(sbindir)"' \
 	    -I$(top_srcdir)/include \
 	    $(LIBOSMOCORE_CFLAGS) \
 	    $(LIBOSMONETIF_CFLAGS) \
 	    $(NULL)
 
+AM_CFLAGS =
+	    -fno-builtin \
+	    -Wall
+
 sgsnemu_LDADD = \
 	@EXEC_LDADD@ \
-	-lgtp -L../gtp ../lib/libmisc.a \
+	-lgtp -L../gtp ../lib/libmisc.la \
 	$(LIBOSMOCORE_LIBS) \
 	$(LIBOSMONETIF_LIBS) \
 	$(NULL)
 
 if ENABLE_GTP_KERNEL
-AM_CFLAGS += \
+AM_CPPFLAGS += \
 	     -DGTP_KERNEL \
 	     $(LIBGTPNL_CFLAGS) \
 	     $(NULL)
 sgsnemu_LDADD += $(LIBGTPNL_LIBS)
 endif
 
-sgsnemu_DEPENDENCIES = ../gtp/libgtp.la ../lib/libmisc.a
+sgsnemu_DEPENDENCIES = ../gtp/libgtp.la ../lib/libmisc.la
 sgsnemu_SOURCES = sgsnemu.c cmdline.c cmdline.h
Index: osmo-ggsn-1.15.0/tests/gtp/Makefile.am
===================================================================
--- osmo-ggsn-1.15.0.orig/tests/gtp/Makefile.am
+++ osmo-ggsn-1.15.0/tests/gtp/Makefile.am
@@ -24,7 +24,7 @@ queue_test_SOURCES = \
 	$(NULL)
 
 gtpie_test_LDADD = \
-	$(top_builddir)/lib/debug.o \
+	$(top_builddir)/lib/debug.lo \
 	$(top_builddir)/gtp/libgtp.la \
 	$(LIBOSMOCORE_LIBS) \
 	$(NULL)
Index: osmo-ggsn-1.15.0/tests/lib/Makefile.am
===================================================================
--- osmo-ggsn-1.15.0.orig/tests/lib/Makefile.am
+++ osmo-ggsn-1.15.0/tests/lib/Makefile.am
@@ -19,7 +19,7 @@ ippool_test_SOURCES = \
 	$(NULL)
 
 ippool_test_LDADD = \
-	$(top_builddir)/lib/libmisc.a \
+	$(top_builddir)/lib/libmisc.la \
 	$(LIBOSMOCORE_LIBS) \
 	$(NULL)
 
@@ -28,6 +28,6 @@ in46a_test_SOURCES = \
 	$(NULL)
 
 in46a_test_LDADD = \
-	$(top_builddir)/lib/libmisc.a \
+	$(top_builddir)/lib/libmisc.la \
 	$(LIBOSMOCORE_LIBS) \
 	$(NULL)
